pkg/integrations/azure_exporter/config.go

```diff
@@ -0,0 +1,270 @@
+
+	// Valid values can be derived from https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported
+	// Required: Root level names ex. Microsoft.DataShare/accounts
+	ResourceType string `yaml:"resource_type"`
+	// Required: Metric in the table for a ResourceType
+	Metrics []string `yaml:"metrics"`
+	// Optional: If not provided Aggregation Type is used for the Metric
+	// Valid values are minimum, maximum, average, total, and count
+	MetricAggregations []string `yaml:"metric_aggregations"`
+
+	// All fields below are optional
+	// Must be an ISO8601 Duration - defaults to PT1M if not specified
+	Timespan             string   `yaml:"timespan"`
+	IncludedDimensions   []string `yaml:"included_dimensions"`
+	IncludedResourceTags []string `yaml:"included_resource_tags"`
+
+	// MetricNamespace is used for ResourceTypes which have multiple levels of metrics
+	// As an example the ResourceType Microsoft.Storage/storageAccounts has metrics for
+	//  Microsoft.Storage/storageAccounts (generic metrics which apply to all storage accounts)
+	//	Microsoft.Storage/storageAccounts/blobServices (generic metrics + metrics which only apply to blob stores)
+	// 	Microsoft.Storage/storageAccounts/fileServices (generic metrics + metrics which only apply to file stores)
+	//	Microsoft.Storage/storageAccounts/queueServices (generic metrics + metrics which only apply to queue stores)
+	//	Microsoft.Storage/storageAccounts/tableServices (generic metrics + metrics which only apply to table stores)
+	// If you want blob store metrics you will need to set
+	//  ResourceType = Microsoft.Storage/storageAccounts
+	//	MetricNamespace = Microsoft.Storage/storageAccounts/blobServices
+	MetricNamespace string `yaml:"metric_namespace"`
+
+	MetricNameTemplate string `yaml:"metric_name_template"`
+	MetricHelpTemplate string `yaml:"metric_help_template"`
+
+	AzureCloudEnvironment string `yaml:"azure_cloud_environment"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config.
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+func (c *Config) InstanceKey(_ string) (string, error) {
+	// Running the integration in v2 as a TypeMultiplex requires the instance key is unique per instance
+	// There's no good unique identifier in the config, so we use a hash instead
+	return getHash(c)
+}
+
+func (c *Config) NewIntegration(l log.Logger) (integrations.Integration, error) {
+	concurrencyConfig := azure_config.Opts{
+		// Necessary to match OSS definition
+		Prober: struct {
+			// Limits the number of subscriptions which can concurrently be sending metric requests - value taken from OSS exporter
+			ConcurrencySubscription int `long:"concurrency.subscription"          env:"CONCURRENCY_SUBSCRIPTION"           description:"Concurrent subscription fetches"                                  default:"5"`
```

Comment:
Out of curiosity: what are these tags used for?

Reply:
Unfortunately, it's necessary to match the OSS struct definition. The `Opts` definition has a bunch of inline structs including this `Prober`, the tags allow the OSS one to do CLI/env parsing (https://github.com/kgeckhart/azure-metrics-exporter/blob/16e25d5c8d3bc9a0e684ae0ac2e8d202fc6e3309/config/opts.go#L34-L39). I was hesitant to touch it in my fork since I didn't want to break the config loading.